ivy_tests/test_ivy/helpers/function_testing.py

```diff
@@ -334,8 +335,11 @@ def test_function(
             if hasattr(ivy.__dict__[fn_name], "out_index")
             else ret
         )
+        # defining random dtype for out array
+        out_dtype = random.choice([ivy.int32, ivy.float32, ivy.complex64])
```

Comment:
The usage of random during testing is prohibited, the tests are no longer deterministic (in a way) which would cause flaky resutls.

Reply:
Got it, then maybe I can try sample from the dtype strategy provided by the test definition